Sources/MixedReality/Microsoft.Psi.MixedReality.Visualization.Windows/ProjectHandsToCamerasTask.cs

```diff
@@ -0,0 +1,152 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license.
+
+namespace Microsoft.Psi.MixedReality.Visualization
+{
+    using System.Collections.Generic;
+    using System.ComponentModel;
+    using System.Linq;
+    using System.Runtime.Serialization;
+    using MathNet.Spatial.Euclidean;
+    using Microsoft.Psi;
+    using Microsoft.Psi.Common.Interpolators;
+    using Microsoft.Psi.Data;
+    using Microsoft.Psi.Imaging;
+    using Microsoft.Psi.MixedReality;
+    using Microsoft.Psi.Spatial.Euclidean;
+    using static Microsoft.Psi.MixedReality.Operators;
+
+    /// <summary>
+    /// Task that projects hands to camera views.
+    /// </summary>
+    [BatchProcessingTask(
+        "Project Hands to Cameras",
+        Description = "Project hand joints based on all available streams of camera image views.",
+        OutputPartitionName = ProjectHandsToCamerasTaskConfiguration.DefaultPartitionName,
+        OutputStoreName = ProjectHandsToCamerasTaskConfiguration.DefaultStoreName)]
+    public class ProjectHandsToCamerasTask : BatchProcessingTask<ProjectHandsToCamerasTaskConfiguration>
+    {
+        /// <inheritdoc/>
+        public override void Run(Pipeline pipeline, SessionImporter sessionImporter, Exporter exporter, ProjectHandsToCamerasTaskConfiguration configuration)
+        {
+            // Get the tracked left and right hand streams
+            var hands = sessionImporter.OpenStream<(Hand Left, Hand Right)>(configuration.HandsStreamName);
+            var leftHand = hands.Select(h => h.Left).Where(h => h is not null && h.IsTracked);
+            var rightHand = hands.Select(h => h.Right).Where(h => h is not null && h.IsTracked);
+
+            foreach (var partition in sessionImporter.PartitionImporters.Values)
+            {
+                foreach (var streamMeta in partition.AvailableStreams)
+                {
+#pragma warning disable SA1101 // Prefix local calls with this
```

Comment:
Is there a reason here for this pragma / not to stick with the stylecop rules?

Reply:
It's a known bug in StyleCop: https://github.com/DotNetAnalyzers/StyleCopAnalyzers/issues/3012